neo/rawio/intanrawio.py

```diff
@@ -53,40 +56,90 @@ class IntanRawIO(BaseRawIO):
        name of the 'rhd' or 'rhs' data file
     """
 
-    extensions = ["rhd", "rhs"]
+    extensions = ["rhd", "rhs", "dat"]
     rawmode = "one-file"
 
     def __init__(self, filename=""):
-        BaseRawIO.__init__(self)
 
+        BaseRawIO.__init__(self)
         self.filename = filename
 
     def _source_name(self):
         return self.filename
 
     def _parse_header(self):
 
+        filename = Path(self.filename)
+        
+        if not filename.exists() or not filename.is_file():
+            raise FileNotFoundError(f"{filename} does not exist")
+
         if self.filename.endswith(".rhs"):
+            self.file_type = 'header-attached'
             self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhs(
                 self.filename
             )
+        # 3 possibilities for rhd files two use a plain header called 'info.rhd' and one format has the header
+        # attached to the actual binary file with data
         elif self.filename.endswith(".rhd"):
+            if filename.name == "info.rhd":
+                # first we have one-file-per-signal which is where one neo stream/file is saved as .dat files
+                if any((filename.parent / file).exists() for file in one_file_per_signal_filenames):
+                    self.file_type = 'one-file-per-signal'
+                    raw_file_paths_dict = create_one_file_per_signal_dict(filename.parent)
+                # then there is one-file-per-channel where each channel in a neo stream is in its own .dat file
+                else:
+                    self.file_type = 'one-file-per-channel'
+                    raw_file_paths_dict = create_one_file_per_channel_dict(filename.parent)
+            # finally the format with the header-attached to the binary file as one giant file
+            else:
+                self.file_type = 'header-attached'
+
             self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhd(
-                self.filename
+                self.filename, self.file_type
             )
 
         # memmap raw data with the complicated structured dtype
-        self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        # if header-attached there is one giant memory-map
+        if self.file_type == "header-attached":
```

Comment:
What do you think of making `_raw_data` a dict so we always index by the available stream_index instead of neding the stream_index_key

Reply:
I'm curious how you see this? I can see how a dict would be good for the file per channel, but how would I make the 'header-attached' be a dict? 

The issue is the user api must take an int. So we would have to incorporate a translation layer. I would have to think about this. If you want to build something off of this you could open a PR on this PR with your idea. Otherwise I'll need to think about this some more.

Reply:
Yeah, in the case of the `header_attached` it does not need to be a dict. Pretty much as we have right now.

The other suggestion would be to use the stream_names as key as conditions like this would be more readable:

```
https://github.com/zm711/python-neo/blob/839060e5b910d878021348de34f50e7bf1ce343e/neo/rawio/intanrawio.py#L114-L118
```

Right now, someone reading the specification would have to remember to which streams these two numbers corresponds two but they would not need two if we were to iterate over the stream_names.

That said, I think this is of little marginal gain and I can prototype in a future PR once we are done with this.

Reply:
That sounds good. After this is merged then you can make some improvements!